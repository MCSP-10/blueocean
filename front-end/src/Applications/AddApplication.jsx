import { useState, useContext } from 'react';
import Button from 'Shared/components/Button/Button';
import { applicationsContext } from 'Applications';
import styles from './AddApplication.module.css';

const AddApplication = (props) => {
    const [company, setCompany] = useState('');
    const [position, setPosition] = useState('');
    const [status, setStatus] = useState(props.status);
    const { createApplication } = useContext(applicationsContext);

    const passValueTo = (callback) => {
        return (e) => {
            callback(e.target.value);
        };
    };
    const onSubmit = async (e) => {
        
        e.preventDefault();   
        if (!company || !position) {
            return;
        }
        const body = { company, job_title: position, status };
        await createApplication(body);

        props.afterSubmit();
    };
    return (
        <>
            <h3 className={styles.title}>Add New Application</h3>
            <hr className={styles.lineBreak}></hr>
            <form onSubmit={onSubmit}>
                <div className={styles.field}>
                    <input
                        type="text"
                        id="company"
                        name="company"
                        onChange={passValueTo(setCompany)}
                        className={styles.inputs}
                        placeholder=" "
                    />
                    <label htmlFor="company" className={styles.labels}>
                        Company Name
                    </label>
                </div>
                <div className={styles.field}>
                    <input
                        type="text"
                        id="job_title"
                        name="job_title"
                        onChange={passValueTo(setPosition)}
                        className={styles.inputs}
                        placeholder=" "
                    />
                    <label htmlFor="job_title" className={styles.labels}>
                        Position
                    </label>
                </div>
                <label className={styles.dropDown}> STATUS: </label>
                <select
                    className={styles.selects}
                    id="status"
                    name="status"
                    defaultValue={status}
                >
                    <option value="Interested">Interested</option>
                    <option value="Applying">Applying</option>
                    <option value="Phone Interview">Phone Interview</option>
                    <option value="Technical Interview">Technical Interview</option>
                    <option value="Final Interview">Final Interview</option>
                    <option value="Offered">Offered</option>
                    <option value="Rejected">Rejected</option>
                </select>
                <br />
                <br />
                <div className={styles.button}>
                    <Button
                        text="Create Application"
                        color="orange"
                        textColor={'chocolate'}
                    />
                </div>
            </form>
        </>
    );
};

export default AddApplication;
