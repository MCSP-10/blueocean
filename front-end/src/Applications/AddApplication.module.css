.inputs {
    margin-top: 1rem;
    border: none;
    background-color: transparent;
    border-bottom: 1px solid var(--darkBlue);
}

.dropDown {
    margin-top: 1rem;
    font-size: 0.9rem;
    color: var(--darkChocolate);
}

.selects {
    font-size: 0.9rem;
    color: var(--darkChocolate);
    margin-top: 1.3rem;
}

.title {
    text-align: center;
    font-size: 1.8rem;
    color: var(--darkBlue);
}

.lineBreak {
    border-top: 1px solid var(--mainOrange);
    margin-top: 0.5rem;
}

.labels {
    color: var(--darkChocolate);
    text-transform: uppercase;
    letter-spacing: 0.05em;
    font-size: 0.8rem;
}

.field {
    display: flex;
    flex-flow: column-reverse;
    margin-bottom: 1em;
}

.labels,
.inputs {
    transition: all 0.2s;
    touch-action: manipulation;
}

.inputs:focus {
    outline: 0;
    border-bottom: 1px solid #666;
}
.inputs:placeholder-shown + .labels {
    cursor: text;
    max-width: 66.66%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    transform-origin: left bottom;
    transform: translate(0, 2.7rem) scale(1.5);
}

::-webkit-input-placeholder {
    opacity: 1;
    transition: inherit;
}

.inputs:focus::-webkit-input-placeholder {
    opacity: 1;
}

.inputs:not(:placeholder-shown) + .labels,
.inputs:focus + .labels {
    transform: translate(0, 1rem) scale(1);
    cursor: pointer;
}

.button {
    display: flex;
    justify-content: center;
}

