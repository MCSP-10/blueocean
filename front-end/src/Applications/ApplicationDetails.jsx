import { useState } from 'react';
import styles from './ApplicationDetails.module.css';
import  Comments  from '../Shared/components/Form/Comments'

const ApplicationDetails = (props) => {
    // console.log(props)
    // const [ comment, makeComment ] = useState('')

    // const postComment = (e) => {
    //     e.preventDefault()
    //     console.log(comment)
    // }

    // const getComments = () => {

    // }



    return(
        <>
        <div className={styles.container}>
            <div className={styles.logo}>
                <img src={props.image}></img>
            </div>
            <div className={styles.details}>
                <h3>Company Name: {props.name}</h3>
                <h3>Title: {props.subText}</h3>
                <h3>Link: {props.url}</h3>
                <h3>Comments: {props.note}</h3>
                {/* <form onSubmit={((e)=>postComment(e))}>
                    <input type='text' onChange={((e)=> makeComment(e.target.value))}></input>
                    <button type='submit'>post comment</button>
                </form> */}
                <Comments/>
            </div>
        </div>
        </>
    );
};

export default ApplicationDetails;