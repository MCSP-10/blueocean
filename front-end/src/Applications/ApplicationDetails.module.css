.container{
    display: flex; 
    flex-direction: row;
    align-items: center;
}

.details{
    padding: 10px;
}

.logo{
    padding:10px;
}