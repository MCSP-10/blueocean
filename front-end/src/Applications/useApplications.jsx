import Api from 'Shared/utils/Api';
import { useQuery } from 'react-query';
import { QueryClient, QueryClientProvider } from 'react-query'
const queryClient = new QueryClient();

const useApplications = () => {
    return (
        <QueryClientProvider client={queryClient}>
            useQuery('applications', Api.applications.getAll);
        </QueryClientProvider>
    )
};

export default useApplications;
