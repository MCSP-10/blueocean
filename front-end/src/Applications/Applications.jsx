import { ApplicationsBoard } from 'Applications';
import { ApplicationsProvider } from 'Applications';
const Applications = () => {
    return (
        <ApplicationsProvider contextSharing={true}>
            <ApplicationsBoard />
        </ApplicationsProvider>
    );
};
export default Applications;
