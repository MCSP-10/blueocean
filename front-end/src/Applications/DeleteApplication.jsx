import React,{ useContext} from 'react';
import { applicationsContext } from 'Applications';
import closeBtn from 'Shared/assets/closeBtn.svg';
import styles from 'Applications/styles/Card.module.css';



const DeleteApplication = (props) => {
    const { deleteApplication } = useContext(applicationsContext);
    const id = props.task_id
    const deleteTask=async ()=>{
        
        await deleteApplication(id)
    }

    return (
        <button onClick={deleteTask}><img className={styles.delete} src={closeBtn}/></button>
    );
};

export default DeleteApplication;
