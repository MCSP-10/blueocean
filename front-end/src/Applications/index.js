import Applications from './Applications.jsx';
import applicationsContext, {
    ApplicationsProvider,
} from './context/applicationsContext.jsx';
import ApplicationsBoard from './ApplicationsBoard/ApplicationsBoard.jsx';
import Column from './ApplicationsBoard/Column.jsx';
import Card from './ApplicationsBoard/Card.jsx';

export { applicationsContext, ApplicationsProvider, ApplicationsBoard, Column, Card };
export default Applications;
