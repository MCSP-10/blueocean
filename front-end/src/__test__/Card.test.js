import React from 'react';
import ReactDOM from 'react-dom';
import { Card } from 'Applications';
import { render, screen, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom";
import { ApplicationsProvider } from 'Applications';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';



//MOCKED RENDER
const renderComponent = ({items}) => {

    return render(
        <ApplicationsProvider>
            <DndProvider backend={HTML5Backend}>
                <Card items={items}/>
            </DndProvider>
        </ApplicationsProvider>
    )
}


describe("Card component", () =>{
    let showModal = false;

    it("Should render a Mapquest job card", () =>{
        
        const component = renderComponent({
            items: { 
                id: 18,
                logo: "https://logo.clearbit.com/Mapquest.com",
                name: "Mapquest",
                subText: "Front End Developer"
            }
        });

        const card = component.getByTestId("app-card");
        expect(card).toBeTruthy();
        expect(card).toMatchSnapshot();
    })

    it("Should open modal app details card", () =>{
        
        ReactDOM.createPortal = jest.fn(() => {
            return (<div data-testid="app-modal">Modal Open</div>);
        });

        const component = renderComponent({
            items: { 
                id: 18,
                logo: "https://logo.clearbit.com/Mapquest.com",
                name: "Mapquest",
                subText: "Front End Developer"
            }
        });
        
        const card = component.getByTestId("app-card");
        expect(fireEvent.click(card)).toBe(true);

        const modal = component.getByTestId("app-modal");
        expect(modal.textContent).toContain("Modal Open");
        // screen.debug();

    })
})

