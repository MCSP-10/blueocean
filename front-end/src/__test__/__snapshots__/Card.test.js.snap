// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Should render a Mapquest job card 1`] = `
<div
  class="component"
  data-testid="app-card"
  draggable="true"
  style="opacity: 1;"
>
  <img
    class="logo"
  />
  <h3
    class="companyName"
  />
  <h3
    class="subText"
  />
  <button
    class="delete"
  >
    <img
      src="https://cdn-icons-png.flaticon.com/128/3221/3221803.png"
    />
  </button>
</div>
`;
